agent/hcp/client/telemetry_config.go

```diff
@@ -86,6 +82,11 @@ func convertAgentTelemetryResponse(ctx context.Context, resp *hcptelemetry.Agent
 		return nil, errInvalidEndpoint
 	}
 
+	var disabled bool
```

Comment:
I feel like we should just set this field TGW side in the response for :
A) Cleaner code in Consul
B) Better control (feels weird that its dependent on the endpoint)
Will discuss and get back to this comment, shouldn't be a big code change, just setting `Disabled` directly from the payload response. But this requires changes TGW side.